Quaver.Shared/Screens/Select/SelectScreen.cs

```diff
@@ -542,5 +580,177 @@ public void SelectRandomMap()
+                    var selectedDifficulty = AvailableMapsets[view.MapsetScrollContainer.SelectedMapsetIndex].Maps[view.DifficultyScrollContainer.SelectedMapIndex];
+
+                    if (selectedDifficulty == null)
+                        return;
+
+                    filePath = selectedMapset.Maps.Count > 1 ? Path.Combine(Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory), selectedDifficulty.Path) : Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            confirmDelete.Confirmation += (sender, confirm) =>
+            {
+                if (!confirm)
+                    return;
+
+                MapDeletingInProgress = true;
+
+                try
+                {
+                    lock (this)
+                    {
+                        // Dispose of the background for the currently selected map.
+                        BackgroundHelper.Background?.Dispose();
+
+                        // Dispose of the currently playing track.
+                        AudioEngine.Track?.Dispose();
+
+                        // Dispose of the selected map.
+                        MapManager.Selected?.Dispose();
+
+                        // Run deletion & Reload in the background.
+                        ThreadScheduler.Run(() => DeleteAndReloadMaps(filePath));
+                    }
+
+                    // Finally show confirmation notification
+                    NotificationManager.Show(NotificationLevel.Success, $"Successfully deleted from Quaver!");
+                }
+                catch (ArgumentOutOfRangeException)
+                {
+                    // Handle error
+                    NotificationManager.Show(NotificationLevel.Error,"An error has occured. Please check runtime.log!");
+                }
+            };
+
+            // Finally show the confirmation dialog that orchestrates the deleting process.
+            DialogManager.Show(confirmDelete);
+        }
+
+        /// <summary>
+        ///     Used to delete a mapset or difficulty, then reload the available maps.
+        /// </summary>
+        /// <param name="path"></param>
+        private void DeleteAndReloadMaps(string path)
+        {
+            var view = View as SelectScreenView;
+            var type = view.ActiveContainer;
+
+            if (view.MapsetScrollContainer.SelectedMapsetIndex < 0)
+                return;
+
+            var preservedMapsetIndex = view.MapsetScrollContainer.SelectedMapsetIndex;
+            var preservedDifficultyIndex = view.DifficultyScrollContainer.SelectedMapIndex;
+
+            try
+            {
+                // delet :amgery:
+                DeletePath(path);
+
+                // Reload cache, reload available mapsets.
+                MapDatabaseCache.Load(false);
```

Comment:
Instead of doing a full load, just delete it from the current state of the cache.